### PR TITLE
feat: Use SQLAlchemy core objects where possible and append only added

### LINKED ISSUE #121: Use SQLAlchemy Table/Column Objects - Names of Tables / Columns
Instead of passing around a string for things like 

https://github.com/MeltanoLabs/target-postgres/blob/main/target_postgres/sinks.py#L83-L87
https://github.com/MeltanoLabs/target-postgres/blob/main/target_postgres/connector.py#L56-L58
https://github.com/MeltanoLabs/target-postgres/blob/main/target_postgres/connector.py#L64-L71

We should use the SQLAlchemy objects we create here https://github.com/MeltanoLabs/target-postgres/blob/main/target_postgres/connector.py#L147 

And reference the related Table / Column Objects. 

This will allow us to punt more work to SQLAlchemy. Issues like 

- https://github.com/MeltanoLabs/target-postgres/issues/119
- https://github.com/MeltanoLabs/target-postgres/issues/52
- https://github.com/MeltanoLabs/target-postgres/pull/59/files (Had to revert because of the craziness here) 
- https://github.com/MeltanoLabs/target-postgres/issues/86 ( I believe if we get SQLAlchemy types around the mapping between Python and pyodbc also gets handled for us avoiding issues like this)

In my opinion were all caused by this issue indirectly and the work to just swap to sqlalchemy isn't that high

### PR DESCRIPTION
**Pull request by visch**
Closes #121, #22, #64, #119, #21, #54

### CODE REVIEW

**Comment by visch** [comment]
This ended up solving more issues than I thought it would. Added them to the main list, just wanted to call it out :)

File: target_postgres/connector.py [hunk 1]
```diff
@@ -53,22 +89,48 @@ def get_sqlalchemy_url(self, config: dict) -> str:
...
+            )
+        if temp_table is True:
+            new_table = sqlalchemy.Table(
+                new_table_name, metadata, *new_columns, prefixes=["TEMPORARY"]
+            )
+        else:
+            new_table = sqlalchemy.Table(new_table_name, metadata, *new_columns)
+        new_table.create(bind=connection)
+        select = sqlalchemy.sql.select(table.columns)
+        insert = Insert(new_table).from_select(names=table.columns, select=select)
```
Comment (visch) [inline target_postgres/connector.py:116]: Major bug here, we only want to clone the metadata of the table. Might be easier to just not worry about the clone step here, and just create a temp table

**Comment by visch** [comment]
Added append only as well

**Comment by pnadolny13** [comment]
@visch I like where this PR is headed, leveraging sqlalchemy for as much as possible is probably a good idea. I've run into a few things while working on target-snowflake that would have benefitted from using sqlalchemy instead of building queries ourseleves.

I didnt see anything worth commenting on from a quick review. I see tests aren't passing yet so feel free to @ me again once this is ready for a final full review.

**Comment by visch** [comment]
> @visch I like where this PR is headed, leveraging sqlalchemy for as much as possible is probably a good idea. I've run into a few things while working on target-snowflake that would have benefitted from using sqlalchemy instead of building queries ourseleves.
> 
> I didnt see anything worth commenting on from a quick review. I see tests aren't passing yet so feel free to @ me again once this is ready for a final full review.

Thanks for taking a quick peak! Sorry about the failure, I missed it after I pushed this up (The tests do pass on Python 3.8+). Fixed it for Python 3.7 and added an issue here for Python 3.7 https://github.com/MeltanoLabs/target-postgres/issues/141

**Comment by visch** [comment]
> @visch I like where this PR is headed, leveraging sqlalchemy for as much as possible is probably a good idea. I've run into a few things while working on target-snowflake that would have benefitted from using sqlalchemy instead of building queries ourseleves.
> 
> I didnt see anything worth commenting on from a quick review. I see tests aren't passing yet so feel free to @ me again once this is ready for a final full review.

Ok they are fixed now @pnadolny13

**Review by pnadolny13 — APPROVED** [review]
@visch this looks good to me!

**Pull request merged**